src/primitives/Sphere.js

```diff
@@ -1,18 +1,29 @@
 import * as THREE from 'three';
 
 import Mesh from './Mesh';
-import { DEFAULT_RADIAL_SEGMENTS } from '../utils/constants';
 
 class Sphere extends Mesh {
-  constructor(color, size = 1) {
-    super();
-    this.geometry = new THREE.SphereGeometry(
-      size,
-      DEFAULT_RADIAL_SEGMENTS,
-      DEFAULT_RADIAL_SEGMENTS
-    );
-    this.material = new THREE.MeshStandardMaterial();
-  }
+    constructor(color , size , widthSegments , heightSegments ) {
+        super();
+        this.geometry = new THREE.SphereGeometry(
+            size,
+            widthSegments,
+            heightSegments
+        );
+        this.material = new THREE.MeshStandardMaterial();
+        this.material.transparent = true;
+        this.mesh = new Mesh();
+    }
+
+    updateOptions(color = "0xffffff", alpha = 1, radius = 0.2, widthSegments = 32, heightSegments = 32) {
```

Comment:
The primitives extend `/src/primitives/Mesh` which has functions for setScale, setColor and setAlpha.
Please refer `src/primitives/Arrow` line 25 for usage.

Reply:
tried to solve the issue I had with the color for sphere by doing an if statement in utils/transform checking if color is a string (thus doing the new THREE.Color) or a dictionnary containing {r,g,b}.

I guess that it's not the best way to do it to stay in line with how you implemented amphion but it clearly shows the issue, it should be reworked of course if it doesn't fit you.